client.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <string>
+#include <vector>
+#include "ChatMessage.hpp" 
+
+
+class ChatClient {
+    boost::asio::io_context _ioContext;
+    boost::asio::ip::tcp::socket _clientSocket;
+    std::string _userName; 
+    bool _isRunning = true;
+
+    static const int bufferSize;
+    static const std::string serverIp;
+    static const int serverPort;
```

Comment:
Same thing here, if you did want this to be shared among all instances, then it's ok